referencing nested optional gives c error: cannot convert 'struct _option_main__IGameObject_ptr' to 'struct _option_main__IGameObject_ptr *'

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 d9b808c, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 5c65e58.d9b808c
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 26100 64-bit
|Processor           |24 cpus, 64bit, little endian, AMD Ryzen 9 3900X 12-Core Processor  
|Memory              |21.16GB/31.9GB
|                    |
|V executable        |C:\v\v.exe
|V last modified time|2025-04-26 19:46:21
|                    |
|V home dir          |OK, value: C:\v
|VMODULES            |OK, value: C:\Users\info\.vmodules
|VTMP                |OK, value: C:\Users\info\AppData\Local\Temp\v_0
|Current working dir |OK, value: D:\vbug2
|                    |
|Git version         |git version 2.42.0.windows.2
|V git status        |weekly.2025.16-51-gd9b808cd-dirty
|.git/config present |true
|                    |
|cc version          |N/A
|gcc version         |N/A
|clang version       |clang version 17.0.1
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |ldd (cygwin) 3.4.7

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg run main.v && D:\vbug2\main.exe`
```v
module main

@[heap]
interface IGameObject {
mut:
	name       string
	parent     ?&IGameObject
	next       ?&IGameObject
	child      ?&IGameObject
	last_child ?&IGameObject
	add_child(mut o IGameObject)
}

@[heap]
struct GameObject implements IGameObject {
mut:
	name       string
	parent     ?&IGameObject
	next       ?&IGameObject
	child      ?&IGameObject
	last_child ?&IGameObject
}

fn (mut gameobject GameObject) add_child(mut o IGameObject) {
	o.parent = gameobject
	if gameobject.last_child != none {
		gameobject.last_child.next = o
	} else {
		gameobject.child = o
	}
	gameobject.last_child = o
}

fn main() {
	mut v1 := &GameObject{
		name: 'v1'
	}
	mut v2 := &GameObject{
		name: 'v2'
	}
	mut v3 := &GameObject{
		name: 'v3'
	}
	v1.add_child(mut v2)
	v1.add_child(mut v3)

	eprintln(v1.child?.next?.name)
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JSTZECKMPD8YPHD2PY5RXYQM.tmp.c:5884: warning: cast between pointer and integer of different size
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JSTZECKMPD8YPHD2PY5RXYQM.tmp.c:5884: error: cannot convert 'struct _option_main__IGameObject_ptr' to 'struct _option_main__IGameObject_ptr *'
...
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01JSTZECKMPD8YPHD2PY5RXYQM.tmp.c:5884: error: cannot convert 'struct _option_main__IGameObject_ptr' to 'struct _option_main__IGameObject_ptr *'
(note: the original output was 5 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

no c error



> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4065,11 +4065,13 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 			g.write('*')
 		}
 		g.write('${tmp_var} = ')
-		if is_ptr {
+		mut needs_addr := false
+		needs_deref := is_ptr && !is_option_unwrap
+		if needs_deref {
 			g.write('*(')
-		}
-		needs_addr := is_option_unwrap && node.expr !in [ast.Ident, ast.PrefixExpr]
-		if is_option_unwrap {
+		} else if is_option_unwrap && !is_ptr {
+			needs_addr = !node.typ.is_ptr()
+				&& node.expr !in [ast.Ident, ast.SelectorExpr, ast.PrefixExpr]
 			if !needs_addr {
 				g.write('&')
 			} else {
@@ -4098,7 +4100,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 			g.write('.')
 		}
 		g.write(field_name)
-		if is_ptr {
+		if needs_deref {
 			g.write(')')
 		}
 		if needs_addr {

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4070,8 +4070,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 		if needs_deref {
 			g.write('*(')
 		} else if is_option_unwrap && !is_ptr {
-			needs_addr = !node.typ.is_ptr()
-				&& node.expr !in [ast.Ident, ast.SelectorExpr, ast.PrefixExpr]
+			needs_addr = node.expr !in [ast.Ident, ast.PrefixExpr]
 			if !needs_addr {
 				g.write('&')
 			} else {

--- a/vlib/v/tests/options/option_nested_selector_test.v
+++ b/vlib/v/tests/options/option_nested_selector_test.v
@@ -1,0 +1,48 @@
+module main
+
+@[heap]
+interface IGameObject {
+mut:
+	name       string
+	parent     ?&IGameObject
+	next       ?&IGameObject
+	child      ?&IGameObject
+	last_child ?&IGameObject
+	add_child(mut o IGameObject)
+}
+
+@[heap]
+struct GameObject implements IGameObject {
+mut:
+	name       string
+	parent     ?&IGameObject
+	next       ?&IGameObject
+	child      ?&IGameObject
+	last_child ?&IGameObject
+}
+
+fn (mut gameobject GameObject) add_child(mut o IGameObject) {
+	o.parent = gameobject
+	if gameobject.last_child != none {
+		gameobject.last_child.next = o
+	} else {
+		gameobject.child = o
+	}
+	gameobject.last_child = o
+}
+
+fn test_main() {
+	mut v1 := &GameObject{
+		name: 'v1'
+	}
+	mut v2 := &GameObject{
+		name: 'v2'
+	}
+	mut v3 := &GameObject{
+		name: 'v3'
+	}
+	v1.add_child(mut v2)
+	v1.add_child(mut v3)
+
+	assert v1.child?.next?.name == 'v3'
+}

